Lagre kartutsnitt-info oftere

jorgkv opened the pull request:
Forsøk på fiks for RO-2583, at kartet hopper til oslo eller sandefjord når man starter ny obs.

Jeg fikk gjenskapt et av tilfellene (men ikke at den hopper til sandefjord), ved å legge til kode som gjorde at gps-posisjonen min oppdaterte seg sjeldnere. Hvis jeg deretter startet registrering av ny obs innen 5 sek etter at appen startet fikk jeg posisjon i Oslo selv om gps-markøren min og opprinnelig utsatt var der hvor jeg var. 

La til `throttleTime(5000)` i denne koden i `geo-position.service.ts`:

```ts
get currentPosition$(): Observable<Position> {
  return this.currentPosition.pipe(
    filter((cp) => cp !== null),
    throttleTime(5000)
  );
}
```

Effekten av det er at det tar litt mer tid før kartet "beveger på seg" og `this.updateMapView();` kalles. På videoene som viser feilen ser det ut som kartet ikke hopper noen ting før ny registrering startes, pluss at det virka som map view ikke ble oppdatert så ofte som det burde blitt i loggene.

Når jeg tester på egen telefon begynner posisjonen min å hoppe litt rundt med en gang, derfor har ikke jeg fått feilen så ofte som Morten, og derfor trengte jeg å legge til `throttleTime`-koden over for å framprovosere feilen.

Som jeg skrev i den ene commit-meldingen så ble trolig `skip(1)` lagt til for å hindre at det ble fyrt av et ekstra søk til apiet under oppstart av applikasjonen. Tror det er bedre at vi håndterer det på en annen måte, feks i `search-criteria.service.ts` eller der hvor selve søket lages, i `home.page.ts`.

jorgkv committed:
Remove skip(1) from map resize subscription

skip(1) ble trolig lagt til for å hindre at det
ble trigget flere søk under oppstart av
applikasjonen. Vi bør heller håndtere det
på en annen måte.

jorgkv committed:
Lytt på zoomend også for å oppdatere map view

Usikker på om det har noen effekt, men så vidt jeg vet bør det ikke ha så mye overhead å lytte på den eventen også. Vi har uansett en devounce(200) som hindrer at videre kode kalles for ofte.

gruble commented:
Tror jeg fikk til å gjenskape det jeg og ved å starte å registrere ny observasjon så fort jeg kunne etter oppstart:
![image](https://github.com/NVE/regObs4/assets/71138449/c4b76f40-4400-4f54-85c0-cf49dcc23d94)
Fikk samme posisjon 3 ganger, men får det ikke til når jeg debugger.
Breddegraden stemmer med unknownMapCenter i settings.ts: [59.911197, 10.741059], men lengdegraden avviker noe, kanskje pga. kartutsnittet?

gruble commented:
Har testet PR'en nå. Får fortsatt samme problem når jeg prøver å registrere en observasjon så fort jeg kan etter oppstart av app. Kanskje det blir lettere å finne ut av det om vi logger litt mer, f.eks. når vi henter posisjon i registreringsbildet?

jorgkv committed:
Allow initialization with lower zoom levels

Should make it clearer to user where he/she is if starting "zoomed out"

jorgkv committed:
Use default location from settings

Also simplifies creation of locationMarker

jorgkv committed:
Remove debounce from extent updates

Also uses updateMapViewOnExtentChange
to not subscribe to events when they are
not used to anything

jorgkv committed:
Clean up initial zoom logic

jorgkv committed:
Change back to zoom level 15

jorgkv committed:
Do not use getter for static observable

gruble reviewed (approved):
Funker bra på web og Android
